packages/store-sync/src/stash/createStorageAdapter.ts

```diff
@@ -23,20 +23,31 @@ export function createStorageAdapter({ stash }: CreateStorageAdapter): StorageAd
       .map((table) => [table.tableId, table]),
   );
 
+  function getRecordId(tableId: Hex, keyTuple: readonly Hex[]): string {
+    return `${tableId}:${concatHex(keyTuple)}`;
+  }
+
   return async function storageAdapter({ logs }: StorageAdapterBlock): Promise<void> {
+    const pendingRecords: Record<string, PendingStashUpdate> = {};
+    const updates: PendingStashUpdate[] = [];
```

Comment:
feels a little weird to do all this tracking here, might be nice if in the future stash would export a util for this like

```
mutate(({ getRecord, setRecord }) => {
  getRecord(...)
  setRecord(...)
});
```

where it internally keeps track of the pending updates/mutations and applies them all at the end of the function execution